src/Tasks/Microsoft.NET.Build.Tasks/ProcessFrameworkReferences.cs

```diff
@@ -778,9 +780,12 @@ enum ToolPackSupport
                 var packNamePattern = knownPack.GetMetadata(packName + "PackNamePattern");
                 var packSupportedRuntimeIdentifiers = knownPack.GetMetadata(packName + "RuntimeIdentifiers").Split(';');
 
+                // When publishing for the RuntimeIdentifier that matches NETCoreSdkPortableRuntimeIdentifier, prefer NETCoreSdkPortableRuntimeIdentifier for the host.
+                // This makes us use a portable ILCompiler when publishing for a portable RID on a non-portable SDK.
+                string hostRuntimeIdentifier = RuntimeIdentifier == NETCoreSdkPortableRuntimeIdentifier ? NETCoreSdkPortableRuntimeIdentifier : NETCoreSdkRuntimeIdentifier;
```

Comment:
@dsplaisted I re-wrote this in a way it could be moved into the `.targets` file.
`NETCoreSdkRuntimeIdentifier` would be passed what `hostRuntimeIdentifier` is set to.

Do you want me to move it to the `.targets` file, or keep it here?